datamol/viz/_substructure.py

```diff
@@ -0,0 +1,104 @@
+import itertools
+
+from typing import Union
+from typing import List
+
+import datamol as dm
+
+from ._viz import to_image
+
+
+def find_bonds_from_atom_list(mol: dm.Mol, atom_indices: List[List[int]]):
+    """Return a list of list of existing bond indices between a list
+    of list of atom indices.
+
+    Args:
+        mol: A molecule.
+        atom_indices: A list of list of atom indices.
+    """
+
+    bonds = []
+
+    for atom_indices_single in atom_indices:
+
+        bonds_single = []
+
+        # Test all possible combinations for existing bonds
+        for a1_idx, a2_idx in itertools.combinations(atom_indices_single, 2):
```

Comment:
On second thought, I am not a huge fan of testing all the combination here. 

Let's use as much rdkit fn as possible.

```python
def atom_list_to_bond(mol, atom_list, bond_as_idx=False):
    amap = {}
    submol = Chem.PathToSubmol(mol, atom_list, atomMap=amap) # <=== here
    amap_reversed = dict((v,k) for k,v in amap.items())
    bonds = []
    # only test bonds of interest not a square matrix
    for bond in submol.GetBonds():
        a1, a2 = bond.GetBeginAtomIdx(), bond.GetEndAtomIdx()
        ori_a1 = amap_reversed[a1]
        ori_a2 = amap_reversed[a2]
        if ori_a1 in atom_list and ori_a2 in atom_list:
            ori_bond = mol.GetBondBetweenAtoms(ori_a1, ori_a2)
            if bond_as_idx:
                bonds.append(ori_bond.GetIdx())
            else:
                bonds.append(bond)
    return bonds
```

Reply:
> On second thought, I am not a huge fan of testing all the combination here.

Honestly, I think it's fine given the size of the graph we are dealing with :-)

But since you already have the function I will replace it by yours (nice catch on `Chem.PathToSubmol`!).

Reply:
Given your below comment (using `pattern` to get the `bond_indices`). I might actually just remove that function.

Do you still want me to add `atom_list_to_bond` for other potential needs? Probably in `dm.mol`.

(you can also do it in your PR if you prefer)

Let me know.

Reply:
With the change below, I don't think it would be necessary (at least in this PR). But maybe we will find some other needs for it eventually.